cmake/External/nccl.cmake

```diff
@@ -75,5 +75,9 @@ if(NOT __NCCL_INCLUDED)
     add_dependencies(__caffe2_nccl ${__NCCL_LIBRARY_DEP})
     target_link_libraries(__caffe2_nccl INTERFACE ${NCCL_LIBRARIES})
     target_include_directories(__caffe2_nccl INTERFACE ${NCCL_INCLUDE_DIRS})
+    # nccl includes calls to shm_open/shm_close and therefore must depend on librt on Linux
+    if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+      target_link_libraries(__caffe2_nccl INTERFACE rt)
```

Comment:
just wondering should this be "PRIVATE rt"?

Reply:
It can not be private, because __caffe2_nccl is a static library therefore all libraries that depends on it need to link on librt)